sql: implement remaining DistSQL-enabled operators in spec exec factory

DrewKimball opened the pull request:
#### sql: remove unused fields and minor refactoring for scanNode init

This commit removes some unused fields from `scanNode` and performs
some minor refactoring for related methods.

Informs #47473

Release note: None

#### sql: refactor embedded scanNode struct

This commit moves some fields from `scanNode` to the new
`fetchPlanningInfo`, which is now embedded in plan nodes that fetch
KVs in place of `scanNode`.

Informs #47473

Release note: None

#### sql: refactor legacy planning

This commit pulls planning-related fields for planNodes that have DistSQL
implementations into `fooPlanningInfo` structs. The corresponding
`DistSQLPlanner` planning methods are now split into a wrapper that
accepts the `planNode`, and the core method that takes the `PlanningInfo`
struct. This prepares the way to implement the remaining DistSQL-enabled
operators in the `distSQLSpecExecFactory`.

As part of the refactor, some unneeded fields and type conversions have
been removed. In addition, some of the `window` planning fields have been
moved to `windowPlanningInfo` instead of being duplicaed in the
`windowFuncHolder` for each window function.

Informs #47473

Release note: None

#### sql: implement remaining DistSQL-enabled operators in DistSQL spec exec factory

This commit adds implementations for the remaining set of DistSQL-enabled
operators in `distSQLSpecExecFactory`.

Informs #47473

Release note: None

#### sql: improve vector insert planning

Vectorized COPY uses an insert with a "literal" values operator as input,
which wraps a `coldata.Batch`. When this is the case, the insert cannot
run as a `planNode`, and must instead use the specialized DistSQL processor.
Previously, the planning logic that determined whether to wrap an insert
`planNode` or create the DistSQL processor was fragile, and required
propagating `isVectorInsert` through the `PlanningCtx`. This commit
plumbs the knowledge that an insert is a "vector insert" from the
optbuilder to DistSQL planning, avoiding the need for the `PlanningCtx`
field. It also clarifies `wrapValuesNode` to explain this case.

Epic: None

Release note: None

#### sql: check preFiltererExpr for distributability

This commit fixes an oversight in the DistSQL planning logic for inverted
filter processors. We now check if the `preFiltererExpr` (if any) can
be distributed.

Epic: None

Release note: None

#### sql: standardize result column naming

This commit changes standardizes the `colinfo.ResultColumns` fields for
`planNode` implementations to be named `columns`.

Epic: None

Release note: None

DrewKimball committed:
sql: remove unused fields and minor refactoring for scanNode init

This commit removes some unused fields from `scanNode` and performs
some minor refactoring for related methods.

Informs #47473

Release note: None

DrewKimball committed:
sql: refactor embedded scanNode struct

This commit moves some fields from `scanNode` to the new
`fetchPlanningInfo`, which is now embedded in plan nodes that fetch
KVs in place of `scanNode`.

Informs #47473

Release note: None

DrewKimball committed:
sql: refactor legacy planning

This commit pulls planning-related fields for planNodes that have DistSQL
implementations into `fooPlanningInfo` structs. The corresponding
`DistSQLPlanner` planning methods are now split into a wrapper that
accepts the `planNode`, and the core method that takes the `PlanningInfo`
struct. This prepares the way to implement the remaining DistSQL-enabled
operators in the `distSQLSpecExecFactory`.

As part of the refactor, some unneeded fields and type conversions have
been removed. In addition, some of the `window` planning fields have been
moved to `windowPlanningInfo` instead of being duplicaed in the
`windowFuncHolder` for each window function.

Informs #47473

Release note: None

DrewKimball committed:
sql: improve vector insert planning

Vectorized COPY uses an insert with a "literal" values operator as input,
which wraps a `coldata.Batch`. When this is the case, the insert cannot
run as a `planNode`, and must instead use the specialized DistSQL processor.
Previously, the planning logic that determined whether to wrap an insert
`planNode` or create the DistSQL processor was fragile, and required
propagating `isVectorInsert` through the `PlanningCtx`. This commit
plumbs the knowledge that an insert is a "vector insert" from the
optbuilder to DistSQL planning, avoiding the need for the `PlanningCtx`
field. It also clarifies `wrapValuesNode` to explain this case.

Epic: None

Release note: None

DrewKimball committed:
sql: implement remaining DistSQL-enabled operators in DistSQL spec exec factory

This commit adds implementations for the remaining set of DistSQL-enabled
operators in `distSQLSpecExecFactory`.

Informs #47473

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/142115)
<!-- Reviewable:end -->

DrewKimball commented:
I'm happy to split this up further into more commits/PRs if that will make for easier review.

yuzefovich reviewed (commented):
Wow, nice work! I only have a handful of comments.

> Reviewed 4 of 4 files at r1, 8 of 8 files at r2, 12 of 12 files at r3, 1 of 1 files at r4, 12 of 12 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/142115#-:-OKP33mS-DSAcwIX8N1l:b-ox8ou6)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @DrewKimball)

___
*[`pkg/sql/union.go` line 61 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/142115#-OKFCLdU1pHOQyp1nzXf:-OKFCLdV0bHu9_kJOiMA:bo7fy3e) ([raw file](https://github.com/cockroachdb/cockroach/blob/7f962d7f0064e7201b06437ec21bf45b4ba13bd7/pkg/sql/union.go#L61)):*
> ```Go
> 
> 	// right and left are the data source operands.
> 	// right is read first, to populate the `emit` field.
> ```

nit `emit` field is long gone, consider removing second part of the sentence.
___
*[`pkg/sql/union.go` line 65 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/142115#-OKFCVnQ-30vcxZbIu6i:-OKFCVnQ-30vcxZbIu6j:b-omyfvs) ([raw file](https://github.com/cockroachdb/cockroach/blob/7f962d7f0064e7201b06437ec21bf45b4ba13bd7/pkg/sql/union.go#L65)):*
> ```Go
> 
> 	// columns contains all the columns from the source, and then
> 	// the columns from the generators.
> ```

nit: what does "generators" refer to? Is there a confusion with project set processor?
___
*[`pkg/sql/opt_exec_factory.go` line 937 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/142115#-OKF7WZ6-tcSh6Zfl_JL:-OKF7WZ79eqytZEmq4W7:b-i1izum) ([raw file](https://github.com/cockroachdb/cockroach/blob/60bb3bbb171ebaead7841dd966795993d145663e/pkg/sql/opt_exec_factory.go#L937)):*
> ```Go
> // Helper function to create a fetchPlanningInfo struct from just a
> // table / index descriptor and requested cols.
> func (ef *execFactory) constructScanForZigzag(
> ```

nit: maybe rename to `constructFetchForZigzag`?
___
*[`pkg/sql/window.go` line 41 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/142115#-OKFAA9Q2qvDt5N0wjiS:-OKFAA9Q2qvDt5N0wjiT:b-i5r9cz) ([raw file](https://github.com/cockroachdb/cockroach/blob/7f962d7f0064e7201b06437ec21bf45b4ba13bd7/pkg/sql/window.go#L41)):*
> ```Go
> type windowPlanningInfo struct {
> 	// columns is the set of result columns.
> 	columns colinfo.ResultColumns
> ```

nit: should this field be moved out of `windowPlanningInfo` struct for symmetry with others?
___
*[`pkg/sql/index_join.go` line 24 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/142115#-OKF99hJE2uMDAReZkgg:-OKF99hJE2uMDAReZkgh:b-83c72b) ([raw file](https://github.com/cockroachdb/cockroach/blob/7f962d7f0064e7201b06437ec21bf45b4ba13bd7/pkg/sql/index_join.go#L24)):*
> ```Go
> 	indexJoinPlanningInfo
> 
> 	// columns are the produced columns, namely the columns fetched from the table
> ```

nit: `s/columns/resultColumns/`.

Also it might be nice to unify this field name across all `planNode`s.
___
*[`pkg/sql/opt/optbuilder/mutation_builder.go` line 1230 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/142115#-OKP2wYj3iK1JU48cmmJ:-OKP2wYkBEvui1bF4BjJ:b-9yttj) ([raw file](https://github.com/cockroachdb/cockroach/blob/294d566ad2dff4d5a02fd8edb6205efd627e0c18/pkg/sql/opt/optbuilder/mutation_builder.go#L1230)):*
> ```Go
> // column metadata needed for the mutation operator.
> //
> // -vectorInsert indicates that the mutation operator is an Insert with a
> ```

nit: add a space: `- vectorInsert`.
___
*[`pkg/sql/distsql_plan_stats.go` line 328 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/142115#-OKF6hyO1oWteBIo9OvJ:-OKF6hyO1oWteBIo9OvK:b23u2vq) ([raw file](https://github.com/cockroachdb/cockroach/blob/60bb3bbb171ebaead7841dd966795993d145663e/pkg/sql/distsql_plan_stats.go#L328)):*
> ```Go
> 
> 	// Initialize a dummy scanNode for the requested statistic.
> 	var scan scanNode
> ```

nit: change in `createPartialStatsPlan` should be in the first commit, not second.
___
*[`pkg/sql/distsql_spec_exec_factory.go` line 158 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/142115#-OKFGsB4BghV_RoUmZoK:-OKFGsB51ikLkWiNzL6G:b-vmq5sw) ([raw file](https://github.com/cockroachdb/cockroach/blob/dae5713abd92c42fa2708503cc8de03cbe018e25/pkg/sql/distsql_spec_exec_factory.go#L158)):*
> ```Go
> 	cols colinfo.ResultColumns, rowCount int,
> ) (exec.Node, error) {
> 	// Special case no columns or no rows.
> ```

nit: maybe move this comment to be on `makeTrivialValues` function itself?
___
*[`pkg/sql/distsql_spec_exec_factory.go` line 197 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/142115#-OKP-9Zd0elEKZP1tM8E:-OKP-9Ze2bO7rJQFCDba:b-mp96y3) ([raw file](https://github.com/cockroachdb/cockroach/blob/dae5713abd92c42fa2708503cc8de03cbe018e25/pkg/sql/distsql_spec_exec_factory.go#L197)):*
> ```Go
> 		physPlan.ResultColumns = cols
> 		return makePlanMaybePhysical(physPlan, planNodesToClose), nil
> 	} else {
> ```

nit: can remove indentation given `return` above.
___
*[`pkg/sql/distsql_spec_exec_factory.go` line 415 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/142115#-OKP-kXr07h7UU4Hg5lX:-OKP-kXr07h7UU4Hg5lY:b-hfktov) ([raw file](https://github.com/cockroachdb/cockroach/blob/dae5713abd92c42fa2708503cc8de03cbe018e25/pkg/sql/distsql_spec_exec_factory.go#L415)):*
> ```Go
> ) (exec.Node, error) {
> 	physPlan, _ := getPhysPlan(input)
> 	recommendation := e.checkExprsAndMaybeMergeLastStage([]tree.TypedExpr{preFiltererExpr}, physPlan)
> ```

It's interesting that we don't check distributability support for `preFiltererExpr` in `checkSupportForPlanNode`, but we probably should for completeness. Consider adding that.
___
*[`pkg/sql/distsql_spec_exec_factory.go` line 1015 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/142115#-OKP1OLYEXl4rtt4ponb:-OKP1OLYEXl4rtt4ponc:bkgucxd) ([raw file](https://github.com/cockroachdb/cockroach/blob/dae5713abd92c42fa2708503cc8de03cbe018e25/pkg/sql/distsql_spec_exec_factory.go#L1015)):*
> ```Go
> 		physPlan.EnsureSingleStreamOnGateway(e.ctx)
> 	}
> 	planCtx := e.getPlanCtx(recommendation)
> ```

In the old code this gets `shouldDistribute` recommendation (unless there is locking or ON expr is not distributed), with a TODO to think about this.

Also I just noticed that we have `shouldDistribute` recommendation in the old code path for zigzag joins while we have `cannotDistribute` in the spec factory, we should fix the latter to match the former.
___
*[`pkg/sql/distsql_physical_planner.go` line 808 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/142115#-OKF7r5L2UjEW0dYjngq:-OKF7r5M9ug1H15yqFib:bo4fd3m) ([raw file](https://github.com/cockroachdb/cockroach/blob/7f962d7f0064e7201b06437ec21bf45b4ba13bd7/pkg/sql/distsql_physical_planner.go#L808)):*
> ```Go
> 		if len(n.partitionIdxs) > 0 {
> 			// If the window has a PARTITION BY clause, then we should distribute the
> 			// execution. TODO(yuzefovich): we might want to be smarter about this and
> ```

nit: TODO should start on a separate line.
___
*[`pkg/sql/distsql_physical_planner.go` line 5150 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/142115#-OKFB8e5Ei6Yf8xF3Udi:-OKFB8e66hLKfNeD6Dze:bwv85sc) ([raw file](https://github.com/cockroachdb/cockroach/blob/7f962d7f0064e7201b06437ec21bf45b4ba13bd7/pkg/sql/distsql_physical_planner.go#L5150)):*
> ```Go
> func (dsp *DistSQLPlanner) planExport(
> 	ctx context.Context, planCtx *PlanningCtx, planInfo *exportPlanningInfo, p *PhysicalPlan,
> ) (err error) {
> ```

nit: I'd avoid the named return parameter here.


<!-- Sent from Reviewable.io -->

DrewKimball committed:
sql: check preFiltererExpr for distributability

This commit fixes an oversight in the DistSQL planning logic for inverted
filter processors. We now check if the `preFiltererExpr` (if any) can
be distributed.

Epic: None

Release note: None

DrewKimball committed:
sql: standardize result column naming

This commit changes standardizes the `colinfo.ResultColumns` fields for
`planNode` implementations to be named `columns`.

Epic: None

Release note: None

DrewKimball reviewed (commented):
TFTR!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/142115#-:-OK_VegCEKNshz481B__:b-814g8k)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @yuzefovich)

___
*[`pkg/sql/distsql_physical_planner.go` line 808 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/142115#-OKF7r5L2UjEW0dYjngq:-OK_U15_D_pnblKt9t-R:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/7f962d7f0064e7201b06437ec21bf45b4ba13bd7/pkg/sql/distsql_physical_planner.go#L808)):*
<details><summary><i>Previously, yuzefovich (Yahor Yuzefovich) wrote…</i></summary><blockquote>

nit: TODO should start on a separate line.
</blockquote></details>

Done.
___
*[`pkg/sql/distsql_physical_planner.go` line 5150 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/142115#-OKFB8e5Ei6Yf8xF3Udi:-OK_U9QBEZ93QGvShFnr:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/7f962d7f0064e7201b06437ec21bf45b4ba13bd7/pkg/sql/distsql_physical_planner.go#L5150)):*
<details><summary><i>Previously, yuzefovich (Yahor Yuzefovich) wrote…</i></summary><blockquote>

nit: I'd avoid the named return parameter here.
</blockquote></details>

Done.
___
*[`pkg/sql/distsql_plan_stats.go` line 328 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/142115#-OKF6hyO1oWteBIo9OvJ:-OK_VMSUBpQcddysykUA:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/60bb3bbb171ebaead7841dd966795993d145663e/pkg/sql/distsql_plan_stats.go#L328)):*
<details><summary><i>Previously, yuzefovich (Yahor Yuzefovich) wrote…</i></summary><blockquote>

nit: change in `createPartialStatsPlan` should be in the first commit, not second.
</blockquote></details>

Done.
___
*[`pkg/sql/distsql_spec_exec_factory.go` line 158 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/142115#-OKFGsB4BghV_RoUmZoK:-OK_UJ0s2-DrewJOa098:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/dae5713abd92c42fa2708503cc8de03cbe018e25/pkg/sql/distsql_spec_exec_factory.go#L158)):*
<details><summary><i>Previously, yuzefovich (Yahor Yuzefovich) wrote…</i></summary><blockquote>

nit: maybe move this comment to be on `makeTrivialValues` function itself?
</blockquote></details>

Done.
___
*[`pkg/sql/distsql_spec_exec_factory.go` line 197 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/142115#-OKP-9Zd0elEKZP1tM8E:-OK_UeNB-_G4UArzKJhn:b-574kmi) ([raw file](https://github.com/cockroachdb/cockroach/blob/dae5713abd92c42fa2708503cc8de03cbe018e25/pkg/sql/distsql_spec_exec_factory.go#L197)):*
<details><summary><i>Previously, yuzefovich (Yahor Yuzefovich) wrote…</i></summary><blockquote>

nit: can remove indentation given `return` above.
</blockquote></details>

Nice, Done.
___
*[`pkg/sql/distsql_spec_exec_factory.go` line 415 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/142115#-OKP-kXr07h7UU4Hg5lX:-OK_DYq_EMFPVN91NTKD:bn22lcv) ([raw file](https://github.com/cockroachdb/cockroach/blob/dae5713abd92c42fa2708503cc8de03cbe018e25/pkg/sql/distsql_spec_exec_factory.go#L415)):*
<details><summary><i>Previously, yuzefovich (Yahor Yuzefovich) wrote…</i></summary><blockquote>

It's interesting that we don't check distributability support for `preFiltererExpr` in `checkSupportForPlanNode`, but we probably should for completeness. Consider adding that.
</blockquote></details>

Done. Added a commit.
___
*[`pkg/sql/distsql_spec_exec_factory.go` line 1015 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/142115#-OKP1OLYEXl4rtt4ponb:-OK_Dc4_CusA-hBQQQ2w:bryrn8j) ([raw file](https://github.com/cockroachdb/cockroach/blob/dae5713abd92c42fa2708503cc8de03cbe018e25/pkg/sql/distsql_spec_exec_factory.go#L1015)):*
<details><summary><i>Previously, yuzefovich (Yahor Yuzefovich) wrote…</i></summary><blockquote>

In the old code this gets `shouldDistribute` recommendation (unless there is locking or ON expr is not distributed), with a TODO to think about this.

Also I just noticed that we have `shouldDistribute` recommendation in the old code path for zigzag joins while we have `cannotDistribute` in the spec factory, we should fix the latter to match the former.
</blockquote></details>

Done - just added it to the spec factory commit.
___
*[`pkg/sql/index_join.go` line 24 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/142115#-OKF99hJE2uMDAReZkgg:-OK_Gz408wHXOoiUSLaV:b-iq3p7w) ([raw file](https://github.com/cockroachdb/cockroach/blob/7f962d7f0064e7201b06437ec21bf45b4ba13bd7/pkg/sql/index_join.go#L24)):*
<details><summary><i>Previously, yuzefovich (Yahor Yuzefovich) wrote…</i></summary><blockquote>

nit: `s/columns/resultColumns/`.

Also it might be nice to unify this field name across all `planNode`s.
</blockquote></details>

Done. I went with `columns` since the vast majority of planNodes use that.
___
*[`pkg/sql/opt_exec_factory.go` line 937 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/142115#-OKF7WZ6-tcSh6Zfl_JL:-OK_SwgREFycLbovYg2-:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/60bb3bbb171ebaead7841dd966795993d145663e/pkg/sql/opt_exec_factory.go#L937)):*
<details><summary><i>Previously, yuzefovich (Yahor Yuzefovich) wrote…</i></summary><blockquote>

nit: maybe rename to `constructFetchForZigzag`?
</blockquote></details>

Done.
___
*[`pkg/sql/union.go` line 61 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/142115#-OKFCLdU1pHOQyp1nzXf:-OK_T8tS8JLlW4n43Hgp:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/7f962d7f0064e7201b06437ec21bf45b4ba13bd7/pkg/sql/union.go#L61)):*
<details><summary><i>Previously, yuzefovich (Yahor Yuzefovich) wrote…</i></summary><blockquote>

nit `emit` field is long gone, consider removing second part of the sentence.
</blockquote></details>

Done.
___
*[`pkg/sql/union.go` line 65 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/142115#-OKFCVnQ-30vcxZbIu6i:-OK_TKGoEaSp5qDz7u_U:b3pqu03) ([raw file](https://github.com/cockroachdb/cockroach/blob/7f962d7f0064e7201b06437ec21bf45b4ba13bd7/pkg/sql/union.go#L65)):*
<details><summary><i>Previously, yuzefovich (Yahor Yuzefovich) wrote…</i></summary><blockquote>

nit: what does "generators" refer to? Is there a confusion with project set processor?
</blockquote></details>

Yes, looks like I copy-pasted something incorrectly. Fixed.
___
*[`pkg/sql/window.go` line 41 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/142115#-OKFAA9Q2qvDt5N0wjiS:-OK_ThbzCHy7l7xDmJEp:b92p7cn) ([raw file](https://github.com/cockroachdb/cockroach/blob/7f962d7f0064e7201b06437ec21bf45b4ba13bd7/pkg/sql/window.go#L41)):*
<details><summary><i>Previously, yuzefovich (Yahor Yuzefovich) wrote…</i></summary><blockquote>

nit: should this field be moved out of `windowPlanningInfo` struct for symmetry with others?
</blockquote></details>

Yes, good point. Done.
___
*[`pkg/sql/opt/optbuilder/mutation_builder.go` line 1230 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/142115#-OKP2wYj3iK1JU48cmmJ:-OK_VV3wC_123BZz3Nj8:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/294d566ad2dff4d5a02fd8edb6205efd627e0c18/pkg/sql/opt/optbuilder/mutation_builder.go#L1230)):*
<details><summary><i>Previously, yuzefovich (Yahor Yuzefovich) wrote…</i></summary><blockquote>

nit: add a space: `- vectorInsert`.
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

yuzefovich reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

Perhaps let's not close #47473 because it's still referenced in the distsql spec factory methods?

> Reviewed 26 of 26 files at r6, 8 of 8 files at r7, 14 of 14 files at r8, 1 of 1 files at r9, 12 of 12 files at r10, 1 of 1 files at r11, 8 of 8 files at r12, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/142115#-:-OKbhlZZAuWhvVTrEkjS:b-izge16)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @DrewKimball)



<!-- Sent from Reviewable.io -->

DrewKimball reviewed (commented):
>Perhaps let's not close [#47473](https://github.com/cockroachdb/cockroach/issues/47473) because it's still referenced in the distsql spec factory methods?

Oops, fixed it.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/142115#-:-OKbr-fu44wP94hlpqo1:bse7wtp)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @yuzefovich)



<!-- Sent from Reviewable.io -->

mgartner reviewed (commented):
Nicely done!!!

> Reviewed 1 of 26 files at r6, 29 of 29 files at r13, 8 of 8 files at r14, 22 of 22 files at r15, 1 of 1 files at r16, 12 of 12 files at r17, 1 of 1 files at r18, 8 of 8 files at r19, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/142115#-:-OLsvTTE7Bbo3GeQMXCm:b-nhzetu)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale)



<!-- Sent from Reviewable.io -->
